addie/step2_handler/generate_sumthing.py

```diff
@@ -48,6 +50,31 @@ def is_sum_inp_file_not_empty(self):
             return False
         return True
 
+    def check_for_write_permissions(self, full_output_file_name):
```

Comment:
The method name doesn't match what the code does. What is a better name?

Reply:
Gotcha, maybe `open_output_file` or `open_output_file_if_writable`?